VisualizationBase/src/Scene.cpp

```diff
@@ -126,6 +142,38 @@ void Scene::removeTopLevelItem(Item* item)
 	scheduleUpdate();
 }
 
+
+ViewItem* Scene::currentViewItem()
+{
+	if (!currentViewItem_)
+	{
+		if (viewItems_.size() == 0)
+			addTopLevelItem(new ViewItem(nullptr, "ProjectView"));
+		else
+			currentViewItem_ = viewItems_.first();
+	}
+	Q_ASSERT(currentViewItem_);
+	return currentViewItem_;
+}
+
+ViewItem* Scene::getViewItem(QString name)
+{
+	for (auto item : viewItems_)
+		if (item->name() == name)
+			return item;
+	return nullptr;
+}
+
+void Scene::switchToView(ViewItem *view)
```

Comment:
There should be an overload that takes a string name and returns true or false if it managed to switch.